### PR TITLE
feat: add plugin icon

### LINKED ISSUE #1168: feat: adding Icons for plugins
# Feature request

## Please describe your feature

 adding Icons for plugins

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues
resolve #1168 

___
### New feature or improvement
- [x] Add default plugin img
image

- [x] Find the corresponding image if possible and replace the default image
image

### CODE REVIEW

**Comment by imjoey** [comment]
@LiteSun nice job. 

> Find the corresponding image if possible and replace the default image.

May I ask that what the selection strategy is for the icons? Which strategy do you prefer? Looking forward to your feedback. Thanks.

**Comment by LiteSun** [comment]
> @LiteSun nice job.
> 
> > Find the corresponding image if possible and replace the default image.
> 
> May I ask that what the selection strategy is for the icons? Which strategy do you prefer? Looking forward to your feedback. Thanks.

The logic is updated in this [commit](https://github.com/apache/apisix-dashboard/pull/1220/commits/e04fb7f7660bc336930eecfd0aa7d2bec78c25c5). Hope this helps you. 😄

File: web/src/components/Plugin/data.tsx [hunk 1]
```diff
@@ -18,149 +18,11 @@ import React from 'react';
...
-  syslog: {
-    category: 'Log',
-    priority: 5,
-  },
-  echo: {
-    category: 'Other',
-    priority: 3,
-  },
+export const PLUGIN_ICON_LIST: Record<string, any> = {
+  prometheus: <IconFont name="iconPrometheus_software_logo" />,
```
Comment (imjoey) [inline web/src/components/Plugin/data.tsx:22]: Hi @LiteSun , are the icon fonts missing?

**Comment by imjoey** [comment]
> The logic is updated in this [commit](https://github.com/apache/apisix-dashboard/pull/1220/commits/e04fb7f7660bc336930eecfd0aa7d2bec78c25c5). Hope this helps you. 😄

@LiteSun , yep, now I know. Thank you very much. I'm curious that if the icon is only for graphical representation, shall we shrink the margins around the icon? I guess that would make the page more compact. Any feedback is welcome. Thanks.

**Comment by LiteSun** [comment]
> > The logic is updated in this [commit](https://github.com/apache/apisix-dashboard/pull/1220/commits/e04fb7f7660bc336930eecfd0aa7d2bec78c25c5). Hope this helps you. 😄
> 
> @LiteSun , yep, now I know. Thank you very much. I'm curious that if the icon is only for graphical representation, shall we shrink the margins around the icon? I guess that would make the page more compact. Any feedback is welcome. Thanks.

I think what you are trying to say is that the plugin cards are too big? This size is OK for me😊, what do you think about the card size and style? @juzhiyuan @liuxiran

**Comment by juzhiyuan** [comment]
Looks fine to me, and we would have a better design in the future.

**Comment by imjoey** [comment]
@nic-chen @juzhiyuan that's right. Let us mainly focus on the functionality for now. Design is a long-term work. 😄

**Comment by liuxiran** [comment]
> > > The logic is updated in this [commit](https://github.com/apache/apisix-dashboard/pull/1220/commits/e04fb7f7660bc336930eecfd0aa7d2bec78c25c5). Hope this helps you. smile
> > 
> > 
> > @LiteSun , yep, now I know. Thank you very much. I'm curious that if the icon is only for graphical representation, shall we shrink the margins around the icon? I guess that would make the page more compact. Any feedback is welcome. Thanks.
> 
> I think what you are trying to say is that the plugin cards are too big? This size is OK for me, what do you think about the card size and style? @juzhiyuan @liuxiran

A little big for me, perhaps It has something to do with the little content in the card, execpt for better degin in the future.

**Comment by juzhiyuan** [comment]
How about using a new PR to improve this?

**Comment by liuxiran** [comment]
> How about using a new PR to improve this?

agree +1

**Review by juzhiyuan — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Comment by imjoey** [comment]
@juzhiyuan Hi, the issue https://github.com/apache/apisix-dashboard/issues/1168 was marked as included in milestone 2.3. Please check that whether this PR requires cherry pick to `v2.3` branch. If yes, please help us attach the `need_backport` label to it. Thank you.

**Comment by juzhiyuan** [comment]
We will check other prs soon

**Comment by imjoey** [comment]
@juzhiyuan  OK, I could do the cherry pick for this for release v2.3.

**Comment by juzhiyuan** [comment]
@imjoey Just done 😂 Thanks a lot!

**Pull request merged**